### PR TITLE
Add Samply Auth

### PR DESCRIPTION
**Pull request by martinbreu**

### CODE REVIEW

File: modules/server/auth/auth.service.tmpl [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ExecStartPre=-/usr/bin/docker rm auth
+ExecStartPre=/usr/bin/docker pull martinbreu/auth:${version}
+ExecStart=/usr/bin/docker run \
+  --name=auth \
+  --network=searchbroker \
+  -e TOMCAT_USERNAME='admin' \
+  -e TOMCAT_PASSWORD='$${TOMCAT_PASS}' \
+  -e POSTGRES_HOST='postgres.openstacklocal' \
+  -e POSTGRES_DB='auth_db' \
+  -e POSTGRES_PORT='5432' \
```
Comment (alexanderkiel) [inline modules/server/auth/auth.service.tmpl]: We should not have such defaults here.

File: modules/server/auth/main.tf [hunk 2]
```diff
@@ -0,0 +1,20 @@
+data "template_file" "auth_service" {
+  template = file("${path.module}/auth.service.tmpl")
+
+  vars = {
+    version = "latest"
+    auth_host = "https://auth.dev.germanbiobanknode.de"
+    mail_host = "out-cloud.mms.t-systems-service.com"
+    mail_port = "25"
```
Comment (alexanderkiel) [inline modules/server/auth/main.tf]: default

File: modules/server/auth/main.tf [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    mail_protocol = "smtp"
```
Comment (alexanderkiel) [inline modules/server/auth/main.tf]: default

**Review by alexanderkiel — CHANGES_REQUESTED** [review]

File: modules/server/auth/main.tf (hunk 2 above)
Comment (martinbreu) [reply modules/server/auth/main.tf]: So mail_port and mail_protocol?
What are the defaults? I looked in searchbroker 2.2.0 but cant find any

File: modules/server/auth/auth.service.tmpl (hunk 1 above)
Comment (martinbreu) [reply modules/server/auth/auth.service.tmpl]: With such you mean POSTGRES_PORT? Should I store the postgres settings elsewhere and access from MDR, Searchbroker and Auth?

Comment (alexanderkiel) [reply modules/server/auth/auth.service.tmpl]: Config of the default Postgres Port 5432 should not be needed here.

File: modules/server/auth/main.tf (hunk 2 above)
Comment (alexanderkiel) [reply modules/server/auth/main.tf]: Yes we should document the defaults in Searchbroker. But usually port 25 is the default smtp port.

**Pull request merged**